stl/inc/type_traits

```diff
@@ -1671,17 +1671,31 @@ struct _Invoke_traits_zero {
 template <class _Callable>
 using _Decltype_invoke_zero = decltype(_STD declval<_Callable>()());
 
+template <class _Void, class _Rx, class _Result>
+struct _Invocable_r_nonvoid : false_type {};
+
+template <class _Ty>
+_NODISCARD _Ty _Returns() noexcept; // not defined

```

Comment:
`_Returns_exactly` might be a slightly more scrutable name here.

Reply:
`_NODISCARD` is probably pointless on a function that isn't defined and therefore cannot be ODR-used.